android/src/app/organicmaps/MwmActivity.java

```diff
@@ -1635,10 +1710,174 @@ public void onLocationUpdated(@NonNull Location location)
   @Override
   public void onCompassUpdated(double north)
   {
+    if (isDestroyed())
```

Comment:
> MwmActivity now checks for Acitivity.isDestroyed() in all callbacks from LocationHelper to avoid any UI operations on already destroyed MwmActivity. This issue caused https://github.com/organicmaps/organicmaps/issues/4550 and https://github.com/organicmaps/organicmaps/issues/4570.

Why is it not possible to unsubscribe the location observer _before_ destroying the activity?

Reply:
Added a comment.